### DISCUSSION TITLE
Status of RP2040

### DISCUSSION THREAD

**Discussion by thestumbler** [modm-io/modm#1238] [Q&A]
Follow-up to a question I asked 7 months ago. How does RP2040 support look these days? RP2350?

The reason it lags behind other ports is simply popularity / demand? Or are there technical issues that prevent it from becoming a first class citizen?

I have a small upcoming project on an RP2040 -- actually an array of RP2040s (same code base). I'd like to try out MODM on the project, but I'll need to interface to an LCD controller, fire up I2S PIO inputs, and talk to a CAN bus transceiver. If RP2040 support is very bare bones, that's a lot of drivers I'll need to port / write in MODM.

**Comment by salkinium** [comment]
modm's strength is mostly in using data-driven code generation as a leverage to compete against ST's HAL, but it's very hard to compete against just 1-2 very well supported chips like the RP2040/RP2350. So the problems are not technical, just a lack of resources and motivation.

I would recommend you check out the official SDK from RPI, since it's really good and there's a very large community to support it.

**Reply by salkinium** [reply]
For PIO there is a WIP PR: https://github.com/modm-io/modm/pull/1212

**Reply by thestumbler** [reply]
Oh that's sad to hear. Yeah, I've used then Pico SDK before, but was really hoping to apply all my MODM experience to the RP2040 on this project. 

While I struggled a bit with the modernness of MODM I really appreciate its adopting C++ when so many other project frameworks are just C. Once you get over the hump, I find c++ is a much better fit for many embedded systems that I work on. 

I might even ditch the RP2040 for an STM chip in the design, since it's still at a preliminary stage.

**Reply by salkinium** [reply]
If you don't need so much ROM/RAM, some of the STM32H5 are a really good deal. Like the [250MHz STM32H533CE with 512kB Flash, 272kB RAM for ~3CHF@10](https://www.digikey.ch/de/products/detail/stmicroelectronics/STM32H533CEU6/22519680).

**Reply by salkinium** [reply]
Oh, interesting, modm doesn't support H5 yet… wtf (but I have a NUCLEO-H503 lying around)

**Reply by thestumbler** [reply]
My project has a bit of a mish mash of interfaces. I need I2S for (very very simple) processing of a stereo audio stream. Also a small LCD / touchscreen, and a library to handle a CANbus controller (like MCP2115 or 2118). Others are standard. 

One reason I was drawn to the RP2040 is that a few companies sell a sandwiched assembly of a touchscreen, an LCD/controller, and an RP2040. I'm not aware of any similar STM32 such modules (there are ESP32 modules, but not sure that'll fly with the customer). 

On the other hand, I'm not 100% crazy about the sandwich approach. I'm going to need an interposer board anyway, and being a sandwich you're stuck with a single supplier. I'm okay with a separate MCU and touchscreen approach. Will look at this chip, thanks.
